test/uk/gov/hmrc/trusts/connectors/DesConnectorSpec.scala

```diff
@@ -115,28 +117,93 @@ class DesConnectorSpec extends BaseConnectorSpec
       "des is returning forbidden response" in {
         val requestBody = Json.stringify(Json.toJson(request))
 
-        stubFor("/trusts/match", requestBody, 409, "{}")
+        stubFor(server, "/trusts/match", requestBody, 409, "{}")
 
         val result = Await.result(connector.checkExistingTrust(request), Duration.Inf)
         result mustBe ServerError
       }
     }
+  }
 
 
-  }
+  ".registerTrust" should {
+
+    "return TRN  " when {
+      "valid request to des register trust." in {
+        val requestBody = Json.stringify(Json.toJson(registrationRequest))
+
+        stubFor(server, "/trusts/registration", requestBody, 200, """{"trn": "XTRN1234567"}""")
+
+        val result = Await.result(connector.registerTrust(registrationRequest), Duration.Inf)
```

Comment:
Can we change this to use ScalaFuture in trait and use 
```
whenReady(future) {
    completed =>
      assert(completed)
}
```

Reply:
Done